DOCS-13703 drop collection sharded environments need cache refreshed (#4312)
mongodb/docs: source/reference/method/db.collection.drop.txt

   collection. The method provides a wrapper around the
   :dbcommand:`drop` command.

   .. note::

      For a sharded cluster, if you use :method:`db.collection.drop()`
      and then create a new collection with the same name, you must
      either:
   
      - Flush the cached routing table on every :binary:`~bin.mongos`
        using :dbcommand:`flushRouterConfig`.

>       - To avoid flushing the cache, you can use
>         :method:`db.collection.remove()` to remove the existing
>         documents and reuse the collection.

Reviewer: Can we please make these bullets more parallel in structure prior to merge?
```suggestion
      - Use :method:`db.collection.remove()` to remove the
        existing documents and reuse the collection. Use this
        approach to avoid flushing the cache.
```

Author: Done per your recommendation.
